### PR TITLE
Migrate deadlock_detector case

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: yeya24 <yb532204897@gmail.com>

### What problem does this PR solve? 

Migrate deadlock_detector case
Fixes https://github.com/pingcap/tipocket/issues/59

### What is changed and how does it work?

Replace pd-ctl with pd http api.

### Check List 

Tests 

 - Unit test
 - E2E test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Go code change
 - Has CI related scripts change
 - Has Terraform scripts change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: pkg/nemesis/leader_shuffle.go [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+	l.shuffleFuncs = []func() error{l.transferLeader, l.transferRegion, l.transferOtherRegionToLeader, l.mergeRegion, l.splitRegion}
+	return l
+}
+
+func (l *leaderShuffler) Invoke(ctx context.Context, node *clusterTypes.Node, args ...interface{}) error {
+	ctx, cancel := context.WithTimeout(context.TODO(), time.Minute*time.Duration(10))
+	defer cancel()
+
+	//pdAddr := fmt.Sprintf("http://%s:%d", node.IP, node.Port)
+	pdAddr := "http://127.0.0.1:2379"
```
Comment (mahjonp) [inline pkg/nemesis/leader_shuffle.go]: ummm

Comment (yeya24) [reply pkg/nemesis/leader_shuffle.go]: Updated

File: tests/pessimistic/deadlock/leader_change.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+type Config struct {
+	DBName           string
+	TableNum         int
+	DeadlockInterval time.Duration
+	DeadlockTimeout  time.Duration
+}
+
+// detectorClient simulates a complete record of financial transactions over the
+// life of a bank (or other company).
+type detectorClient struct {
```
Comment (mahjonp) [inline tests/pessimistic/deadlock/leader_change.go]: ahhh, we have so many bank cases.

**Comment by mahjonp** [comment]
almost LGTM

**Comment by yeya24** [comment]
@mahjonp Update. PTAL.

**Pull request merged**